error initialized to nil

The following program `sample.go` triggers a panic:

```go
package main

func main() {
    err := error(nil)
    println(err == nil)
}
```

Expected result:
```console
$ go run ./sample.go
true
```

Got:
```console
$ yaegi ./sample.go
scope.go:158: nil reflect type
panic: nil reflect type [recovered]
        panic: /home/marc/tmp/issue-e1.go:4:2: CFG post-order panic: nil reflect type

goroutine 1 [running]:
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2.1(0xc000272800)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:369 +0xba
panic(0xc9eaa0, 0xc000078450)
        /usr/lib/go/src/runtime/panic.go:679 +0x1b2
log.Panic(0xc0001b7278, 0x1, 0x1)
        /usr/lib/go/src/log/log.go:338 +0xac
github.com/containous/yaegi/interp.(*scope).add(0xc00008a6c0, 0xc000189ba0, 0xc000272a00)
        /home/marc/go/src/github.com/containous/yaegi/interp/scope.go:158 +0x173
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2(0xc000272800)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:427 +0x7bf6
github.com/containous/yaegi/interp.(*node).Walk(0xc000272800, 0xc0001b7d00, 0xc0001b7d28)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:171 +0xaf
github.com/containous/yaegi/interp.(*node).Walk(0xc000272700, 0xc0001b7d00, 0xc0001b7d28)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc000272200, 0xc0001b7d00, 0xc0001b7d28)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc000272000, 0xc0001b7d00, 0xc0001b7d28)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc000260000, 0xc000272000, 0x7ffd620a969f, 0x1a, 0x7ffd620a969f, 0x1a, 0x0, 0x0, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:51 +0x16d
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000260000, 0xc00026c000, 0x46, 0x246, 0xc00026c000, 0x46, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:322 +0x228
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42e
```

--- a/_test/type15.go
+++ b/_test/type15.go
@@ -1,0 +1,9 @@
+package main
+
+func main() {
+	err := error(nil)
+	println(err == nil)
+}
+
+// Output:
+// true

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -743,7 +743,7 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 				if isInt(n.child[0].typ.TypeOf()) && n.child[1].kind == basicLit && isFloat(n.child[1].typ.TypeOf()) {
 					err = n.cfgErrorf("truncated to integer")
 				}
-				if isInterface(n.child[0].typ) {
+				if isInterface(n.child[0].typ) && !n.child[1].isNil() {
 					// Convert to interface: just check that all required methods are defined by concrete type.
 					c0, c1 := n.child[0], n.child[1]
 					if !c1.typ.implements(c0.typ) {
@@ -1694,6 +1694,9 @@ func (n *node) isNatural() bool {
 	}
 	return false
 }
+
+// isNil returns true if node is a literal nil value, false otherwise
+func (n *node) isNil() bool { return n.kind == basicLit && !n.rval.IsValid() }
 
 // fieldType returns the nth parameter field node (type) of a fieldList node
 func (n *node) fieldType(m int) *node {

--- a/interp/run.go
+++ b/interp/run.go
@@ -216,7 +216,7 @@ func convert(n *node) {
 	typ := n.child[0].typ.TypeOf()
 	next := getExec(n.tnext)
 
-	if c.kind == basicLit && !c.rval.IsValid() { // convert nil to type
+	if c.isNil() { // convert nil to type
 		n.exec = func(f *frame) bltn {
 			dest(f).Set(reflect.New(typ).Elem())
 			return next

